pkg/providers/nutanix/template.go

```diff
@@ -346,10 +346,53 @@ func buildTemplateMapCP(
 	return values, nil
 }
 
+func calcFailureDomainReplicas(workerNodeGroupConfiguration v1alpha1.WorkerNodeGroupConfiguration, failureDomains []v1alpha1.NutanixDatacenterFailureDomain) map[string]int {
+	replicasPerFailureDomain := make(map[string]int)
+	failureDomainCount := len(failureDomains)
+
+	if workerNodeGroupConfiguration.AutoScalingConfiguration != nil {
+		return replicasPerFailureDomain
+	}
+
+	if failureDomainCount == 0 {
+		return replicasPerFailureDomain
+	}
+
+	workerNodeGroupCount := failureDomainCount
+	if workerNodeGroupConfiguration.Count != nil {
+		workerNodeGroupCount = int(*workerNodeGroupConfiguration.Count)
+	}
+
+	minCount := int(workerNodeGroupCount / failureDomainCount)
+
+	for i := 0; i < len(failureDomains); i++ {
+		replicasPerFailureDomain[failureDomains[i].Name] = minCount
+	}
+	replicasPerFailureDomain[failureDomains[0].Name] = workerNodeGroupCount - (failureDomainCount-1)*minCount
```

Comment:
do we support the case where `workerNodeGroupCount` < `failureDomainCount`?
`minCount` in that case would be 0 and the first failure domain will get all the replicas. That's not what we expect right? Should we just block that case through validations?

Reply:
Fixed: https://github.com/aws/eks-anywhere/pull/8837/files#diff-58957f424e4554682472c6e2df3353da80d6baf33d1f63ad79b319d57c241b17R184